### DISCUSSION TITLE
How to execute a Windows-based batch file (.bat file) or an installer for a tenant environment

### DISCUSSION THREAD

**Discussion by tomg-formpak** [awslabs/aws-saas-boost#544] [Q&A]
What are suggested ways to execute a Windows-based batch file (.bat file) or an installer for a tenant environment?

Assume we have 5 tenants onboarded and live in our production environment. Now how to run a window batch file (.bat file) like a Talend job with required files or a Windows installer file for a specific tenant's environment.

I was thinking something like uploading the resources (executable file) in that tenant-specific s3 bucket, which acts as a trigger for the event and then executes AWS Lambda to perform execution of the Windows-based batch file or Windows installer or even a patch file, with required parameters using AWS Systems Manager Run Command or AWS Step Functions.

Please suggest this since I am not sure how this changes or integrates with the ECS cluster ECS task or the application container or RDS instance for the DB-related changes.

**Comment by brtrvn** [comment]
@tomg-formpak I'd look at ECS Exec as a way to invoke commands on a running task.
https://aws.amazon.com/blogs/containers/introducing-amazon-ecs-exec-to-access-your-windows-containers-on-amazon-ec2/
